src/js/stores/MesosStateStore.js

```diff
@@ -73,20 +67,34 @@ class MesosStateStore extends GetSetBaseStore {
 
   subscribe() {
     const mesosStream = container.get(MesosStreamType);
-    const parsers = pipe(...Object.values(mesosStreamParsers));
     const getMasterRequest = request(
       { type: "GET_MASTER" },
       "/mesos/api/v1?get_master"
     ).retry(3);
 
-    this.stream = mesosStream
+    const parsers = pipe(...Object.values(mesosStreamParsers));
+    const dataStream = mesosStream
       .merge(getMasterRequest)
       .map(message => parsers(this.get("lastMesosState"), JSON.parse(message)))
       .map(MesosStateUtil.flagMarathonTasks)
       .do(state => {
         CompositeState.addState(state);
         this.setState(state);
-      })
+      });
+
+    const waitStream = getMasterRequest.zip(mesosStream.take(1));
+    const eventTriggerStream = dataStream.merge(
+      // Lots of things in DCOS UI rely on the MesosStateStore emitting
+      // MESOS_STATE_CHANGE events but since we're switching to the stream there will
+      // be no events coming. To avoid a bigger refactoring I decided to go with
+      // a fake emitter that starts emitting events once the initial state came through
+      // TODO: https://jira.mesosphere.com/browse/DCOS-18277
+      Rx.Observable.interval(Config.getRefreshRate())
+    );
+
+    waitStream
+      .concat(eventTriggerStream)
+      .debounceTime(1000)
```

Comment:
Please calculate the debounce time as `Config.getRefreshRate() * .5` and document the min/max frequency of events.